src/core/mac/mac.cpp

```diff
@@ -507,13 +509,65 @@ otError Mac::SetShortAddress(ShortAddress aShortAddress)
     return OT_ERROR_NONE;
 }
 
-otError Mac::SetChannel(uint8_t aChannel)
+otError Mac::SetPanChannel(uint8_t aChannel)
 {
     otError error = OT_ERROR_NONE;
 
     VerifyOrExit(OT_RADIO_CHANNEL_MIN <= aChannel && aChannel <= OT_RADIO_CHANNEL_MAX, error = OT_ERROR_INVALID_ARGS);
 
-    mChannel = aChannel;
+    mPanChannel = aChannel;
+
+    VerifyOrExit(!mRadioChannelAcquisitionId);
+
+    mRadioChannel = mPanChannel;
+
+    UpdateIdleMode();
+
+exit:
+    return error;
+}
+
+otError Mac::SetRadioChannel(uint16_t aAcquisitionId, uint8_t aChannel)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(OT_RADIO_CHANNEL_MIN <= aChannel && aChannel <= OT_RADIO_CHANNEL_MAX, error = OT_ERROR_INVALID_ARGS);
+    VerifyOrExit(mRadioChannelAcquisitionId && aAcquisitionId == mRadioChannelAcquisitionId,
+                 error = OT_ERROR_INVALID_STATE);
+
+    mRadioChannel = aChannel;
+
+    UpdateIdleMode();
+
+exit:
+    return error;
+}
+
+otError Mac::AcquireRadioChannel(uint16_t *aAcquisitionId)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(aAcquisitionId != NULL, error = OT_ERROR_INVALID_ARGS);
+    VerifyOrExit(!mRadioChannelAcquisitionId, error = OT_ERROR_INVALID_STATE);
+
+    mRadioChannelAcquisitionId =
+        static_cast<uint16_t>(Random::GetUint32InRange(1, (1U << (sizeof(mRadioChannelAcquisitionId) * 8)) - 1));
```

Comment:
suggestion: how about using `Random::GetUint16InRange`? I guess  the `1U << (sizeof()...)` constant may not work? Maybe we can just use `0xffff` directly or define a `kMaxRadioChannelAcquisitionId` constant?

Reply:
will fix.